pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -802,6 +835,25 @@ func (lrw *logicalReplicationWriterProcessor) flushBuffer(
 		stats.Add(lrw.bhStats[i])
 	}
 
+	// If any chunk flush failed when run as a batch but succeeded when re-run
+	// de-batched, we may be seeing keys/revision combinations that do not benefit
+	// from batching and where doing so, only to fail and need to try again, is
+	// indeed increasing our cost, so we may want to disable batching in future
+	// flushes, at least for some time. If the workload continues to emit flushes
+	// that encounter this, we may want to disable batching for a longer period,
+	// so we ratchet up the number of flushes we will skip per flush that hits a
+	// batching failure using skipBatchingStep. We do however cap this backoff so
+	// that we still eventually will try another batched flush in case the traffic
+	// has changed at a later point.
+	if stats.debatchResolvedErrs > 0 {
```

Comment:
we may need some observability around how often we are actually running multi row batches-- perhaps count of multirow batches in the debug table? like it would be good to know if this backoff policy is too restrictive and leads us to never batch.

Reply:
we already have batch count in the vtable